src/SelectField/SelectField.js

```diff
@@ -0,0 +1,212 @@
+     *
+     * @param {object} event TouchTap event targeting the menu item
+     * that was selected.
+     * @param {number} key The index of the selected menu item.
+     * @param {any} payload The `value` prop of the selected menu item.
+     */
+    onChange: PropTypes.func,
+    /** @ignore */
+    onFocus: PropTypes.func,
+    /**
+     * Whether the label should be displayed as required (asterisk).
+     */
+    required: PropTypes.bool,
+    /**
+     * Type of the input element. It should be a valid HTML5 input type.
+     */
+    type: PropTypes.string,
+    /**
+     * The input value, required for a controlled component.
+     */
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  };
+
+  static defaultProps = {
+    disabled: false,
+  };
+
+  static contextTypes = {
+    styleManager: customPropTypes.muiRequired,
+  };
+
+  state = {
+    anchorEl: undefined,
+    open: false,
+    selectedIndex: undefined,
+  };
+
+  handleMouseDown = (event) => event.preventDefault();
+
+  handleClick = (event) => this.setState({ open: true, anchorEl: event.currentTarget });
+
+  handleRequestClose = () => this.setState({ open: false })
+
+  handleItemClick = (event, index, value) => {
+    event.persist();
+    this.setState({
+      open: false,
+    }, () => {
+      if (this.props.onChange) {
+        this.props.onChange(event, index, value);
+      }
+
+      // this.close(Events.isKeyboard(event));
+    });
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      disabled,
+      error,
+      hideLabel,
+      inputClassName,
+      inputProps,
+      label,
+      labelClassName,
+      menuClassName,
+      menuProps,
+      required,
+      type,
+      value,
+      ...other
+    } = this.props;
+
+    const classes = this.context.styleManager.render(styleSheet);
+    const initialShrink = value !== '' && typeof value !== 'undefined';
+
+    return (
+      <FormControl
+        className={className}
+        error={error}
+        required={required}
+        {...other}
+      >
+        {label && !(hideLabel && value) && (
+          <InputLabel className={labelClassName} shrink={initialShrink}>
+            {label}
+          </InputLabel>
+        )}
+        <Input
+          className={inputClassName}
+          value={value}
+          type={type}
+          disabled={disabled}
+          onMouseDown={this.handleMouseDown}
+          onClick={this.handleClick}
+          component={SelectFieldInput}
+          label={label}
+          options={children}
+          {...inputProps}
+        />
+        <Menu
+          anchorEl={this.state.anchorEl}
+          className={classNames(classes.menu, menuClassName)}
+          open={this.state.open}
+          onRequestClose={this.handleRequestClose}
+          {...menuProps}
+        >
+          {React.Children.map(children, (child, index) =>
+            React.cloneElement(child, {
+              selected: value === child.props.value,
```

Comment:
It worth discussion, cause previous API supported this http://www.material-ui.com/#/components/select-field

And it would be another breaking change.

Anyways it's easy to write wrapper over it.